src/StopwatchPlugin.php

```diff
@@ -39,11 +40,11 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
         $this->stopwatch->start($eventName, self::CATEGORY);
 
         return $next($request)->then(function (ResponseInterface $response) use ($eventName) {
-            $this->stopwatch->stop($eventName, self::CATEGORY);
+            $event = $this->stopwatch->stop($eventName);
 
-            return $response;
+            return $response->withHeader(self::HEADER, $event->getDuration());
         }, function (Exception $exception) use ($eventName) {
-            $this->stopwatch->stop($eventName, self::CATEGORY);
+            $this->stopwatch->stop($eventName);
```

Comment:
indeed, the category parameter seems unneeded according to https://github.com/symfony/stopwatch/blob/master/Stopwatch.php